server/stream.go

```diff
@@ -36,6 +36,14 @@ import (
 	"github.com/nats-io/nuid"
 )
 
+// StreamRequest is used to create or update a stream.
+type StreamRequest struct {
```

Comment:
We have `CreateConsumerRequest` which seems very clear, where `StreamRequest` seems ambigious - we have many other kinds of stream request other than create/update.  So maybe a better for this would be nice.